### PR TITLE
Re-export `futures_lite` in `bevy_tasks`

### LINKED ISSUE #10664: Re-export `poll_once` from Bevy
## What problem does this solve or what need does it fill?

In order to perform async tasks users need to use `futures-lite`, see [`async_compute`](https://github.com/bevyengine/bevy/blob/main/examples/async_tasks/async_compute.rs#L91C51-L91C60) example.

## What solution would you like?

It would be great to re-export `future` module from `bevy::tasks` to avoid adding `futures-lite` explicitly in `Cargo.toml`. And I would probably remove `block_on` from the prelude.

### PR DESCRIPTION
**Pull request by BD103**
# Objective

- Fixes #10664

## Solution

- `pub use futures_lite as future` :)

---

## Changelog

- Made `futures_lite` available as `future` in the `bevy_tasks` module.

## Migration Guide

- Remove `futures_lite` from `Cargo.toml`.

```diff
[dependencies]
bevy = "0.12.0"
- futures-lite = "1.4.0"
```

- Replace `futures_lite` imports with `bevy::tasks::future`.

```diff
- use futures_lite::poll_once;
+ use bevy::tasks::future::poll_once;
```

### CODE REVIEW

**Comment by BD103** [comment]
Some other things I want to bring up:

- The original issue suggests removing `block_on` from `bevy_tasks`. Right now, `block_on` can either be from `futures_lite` or `async_io`. Should I leave it how it is, or do you have any other suggestions?

https://github.com/bevyengine/bevy/blob/7ff61a8dc9b7a54284d0c864d1908271eeb2cc88/crates/bevy_tasks/src/lib.rs#L30-L33

- `future` is a really generic module name. Should I keep the re-export as `futures_lite` instead?

---

I also need to update the related examples, so please don't merge this yet. :)

**Comment by alice-i-cecile** [comment]
> The original issue suggests removing block_on from bevy_tasks. Right now, block_on can either be from futures_lite or async_io. Should I leave it how it is, or do you have any other suggestions?

Leave it as is IMO.

**Review by alice-i-cecile — APPROVED** [review]

File: examples/async_tasks/async_compute.rs [hunk 1]
```diff
@@ -3,9 +3,8 @@
 
 use bevy::{
     prelude::*,
-    tasks::{block_on, AsyncComputeTaskPool, Task},
+    tasks::{block_on, future::future, AsyncComputeTaskPool, Task},
```
[the remark this answers is not in the document]
Comment (BD103) [reply examples/async_tasks/async_compute.rs:6]: I agree, let me commit fix that.

[the remark this answers is not in the document]
Comment (Shatur) [reply examples/async_tasks/async_compute.rs:6]: Looks like you forgot to change the example.

**Review by Shatur — APPROVED** [review]

**Pull request merged**